docker-compose.yml

```diff
@@ -0,0 +1,45 @@
+
+version: "3"
+
+services:
+
+  fuseki:
+    build: fuseki
+    ports:
+      - 3030:3030
+    environment:
+      - ADMIN_PASSWORD=password
+      - ENABLE_DATA_WRITE=true
+      - ENABLE_UPDATE=true
+      - ENABLE_UPLOAD=true
+  datacube:
+    build: .
+    ports:
+      - "5678:5678"
+      - "5679:5679"
+    command: nodemon --exec node --inspect=0.0.0.0:5679 -r ts-node/register --inspect index.ts
+    depends_on:
+      - fuseki
+    environment:
+      PORT: 5678
+      BASE_URI: http://localhost:5678/
+      SPARQL_ENDPOINT: http://fuseki:3030/data-cube-e2e/query
+      SPARQL_UPDATE_ENDPOINT: http://fuseki:3030/data-cube-e2e/update
+      READ_MODEL_SPARQL_ENDPOINT: http://fuseki:3030/data-cube-read/query
+      READ_MODEL_SPARQL_UPDATE_ENDPOINT: http://fuseki:3030/data-cube-read/update
+      STORAGE_FILESYSTEM_PATH: /uploads
+    volumes:
+      - uploads:/uploads
+      - ./api:/api
+  analyzer:
+    image: "hydrofoil/hydra-analyser"
+    network_mode: "host"
+    environment:
+      ENTRYPOINT_URL: "http://localhost:5678/"
+  e2e-tests:
+    build: api/test
+    network_mode: "host"
+    environment:
+      BASE_URI: "http://localhost:5678/"
```

Comment:
The api is addressed at `localhost:5678` and not `datacube:5678` because otherwise the resource identifiers are going to be a pain. (unless we want to edit the `hosts` file)

Reply:
I'm not sure I understand this, did you have issues using `datacube:5678`?

> by default, any service can reach any other service at that service’s name

so the name `datacube` should resolve to the datacube service IP without having to manually change the hostsfile

Reply:
`datacube:5678` is not accessible from host's network which still has to use `localhost`. This causes a problem with resource identifiers, because we only have a single namespace, as defined in the API's startup parameters.

Otherwise sure, it would work for the test runner.

Reply:
Ah right, I missed the `network_mode: "host"`. Do you really need it?

Reply:
Yes, when using `localhost:5678` as the base uri